src/components/LineChart/index.js

```diff
@@ -1,300 +1,118 @@
-    if (!(x === prevx && k === prevk)) {
-      this.selection.call(this.props.zoom.transform, this.props.transformation);
-    }
+const Wrapper = styled.div`
+  display: grid;
+  grid-template-columns: 1fr auto;
 
-    if (prevProps.config.zoomable != this.props.config.zoomable) {
-      this.syncZoomingState();
-    }
+  .handle {
+    width: 6px;
+    fill: rgb(0, 72, 125);
   }
+`;
 
-  syncZoomingState = () => {
-    if (this.isZoomable()) {
-      this.selection.call(this.props.zoom.on('zoom', this.zoomed));
-    } else {
-      this.selection.on('.zoom', null);
-    }
+class LineChartComponent extends Component {
+  static propTypes = {
+    zoomable: PropTypes.bool,
+    size: PropTypes.shape({
+      width: PropTypes.number.isRequired,
+      height: PropTypes.number,
+    }).isRequired,
+    series: PropTypes.array.isRequired,
+    crosshair: PropTypes.bool,
+    baseDomain: PropTypes.arrayOf(PropTypes.number).isRequired,
+    subDomain: PropTypes.arrayOf(PropTypes.number).isRequired,
+    yAxisWidth: PropTypes.number.isRequired,
+    contextChartHeight: PropTypes.number,
   };
 
-  zoomed = () => {
-    const t = d3.event.transform;
-    const scale = this.props.xScale;
-    const newScale = t.rescaleX(scale);
-    const dd = newScale.domain().map(p => p.getTime());
-    return this.props.subDomainChanged(dd);
+  static defaultProps = {
+    zoomable: true,
+    contextChartHeight: 0.1,
+    crosshair: true,
   };
 
-  calculateDomainFromData = (data, yAccessor) => {
-    const extent = d3.extent(data, yAccessor);
-    const diff = extent[1] - extent[0];
-    if (Math.abs(diff) < 1e-3) {
-      return [1 / 2 * extent[0], 3 / 2 * extent[0]];
-    }
-    return [extent[0] - diff * 0.025, extent[1] + diff * 0.025];
-  };
+  state = {};
 
   render() {
     const {
-      yAxis,
-      xAxis,
+      size: { width },
       series,
-      height,
-      width,
-      heightPct,
-      offsetY,
-      subXScale: xScale,
-      rescaleY,
-      colors,
-      onMouseMove,
-      onClickAnnotation,
-      crosshairs,
-      margin,
-      hiddenSeries,
-      annotations,
-      config,
-      strokeWidths,
+      yAxisWidth,
+      contextChartHeight,
+      baseDomain,
+      subDomain,
+      crosshair,
     } = this.props;
-    const effectiveHeight = height * heightPct;
-    const { linex, liney } = this.state;
-    let yAxisDisplayMode = 'ALL';
-    if (config.yAxis && config.yAxis.display) {
-      yAxisDisplayMode = config.yAxis.display;
-    }
-    const showAxes = yAxisDisplayMode === 'ALL';
+    const height = 800;
+    const visibleSeries = series.filter(s => !s.hidden);
+    const axisCollectionWidth = yAxisWidth * visibleSeries.length;
+    const contextHeight = contextChartHeight * height;
+    const lineCollectionHeight = height - contextHeight;
+    const lineCollectionWidth = width - axisCollectionWidth;
+    const subXScale = createXScale(subDomain, lineCollectionWidth);
```

Comment:
It seems all the child components calculate createXScale inside taking width and domain as props, but Axis takes already calculated scale as a parameter, so I would either move it down to the Axis component or calculate it only once in the LineChat upper level and provide all the child components with the value as a prop.

Reply:
Yep -- `Axis` is still legacy.